### ISSUE TITLE
Object.dirname is deprecated.

### ISSUE THREAD

**Issue by blimmer** [AtomLinter/linter-codeclimate#49]
Argument to `path.dirname` must be a string
```
Object.dirname (/Applications/Atom.app/Contents/Resources/app.asar/src/electron-shims.js:9:10)
Object.lint (/Users/blimmer/.atom/packages/linter-codeclimate/lib/linter-codeclimate.coffee:113:29)
promises.push.Promise.then._this.emitter.emit.linter (/Users/blimmer/.atom/packages/linter/lib/linter-registry.js:95:28)
<unknown> (/Users/blimmer/.atom/packages/linter/lib/linter-registry.js:94:25)
Set.forEach (native)
LinterRegistry.lint (/Users/blimmer/.atom/packages/linter/lib/linter-registry.js:92:20)
```

**Comment by maxjacobson** [comment]
Hi @blimmer thanks for the issue. I'm trying to reproduce, but not currently seeing that output anywhere. Will you help show me where to look?

**Comment by blimmer** [comment]
Definitely - I think the first step will be to make sure you're up to date on Atom. I'm seeing this on version 1.13.0.

Then, it's in the bottom right of my editor:

![](https://i.imgur.com/ZNpLu6U.png)

![](https://i.imgur.com/fshlFAs.png)

**Comment by maxjacobson** [comment]
@blimmer hm, still not seeing it:

![screen shot 2017-01-16 at 11 50 10 am](https://cloud.githubusercontent.com/assets/1421211/21991787/fce15562-dbe1-11e6-9a2f-ef2cb65f5932.png)

I'm using:

- macOS
- Atom 1.13.0
- linter 1.11.21
- linter-codeclimate 0.2.1

**Comment by blimmer** [comment]
Ah, upon looking at my atom deps, I bet it's [this package](https://github.com/atom/deprecation-cop) that's doing it.

**Comment by maxjacobson** [comment]
@blimmer oh interesting. It looks like this one is on by default, because I seem to be running that as well (and still no deprecation output for me)

**Comment by blimmer** [comment]
hmm, not sure. all of my versions exactly match yours.

**Comment by Arcanemagus** [comment]
I'm betting that [this](https://github.com/AtomLinter/linter-codeclimate/blob/cc88e576bcf4809cc2fd96e8d7cb7c87a73ad9c8/lib/linter-codeclimate.coffee#L75) is unable to find a configuration file for you @blimmer, which is leading to a `null` being sent to `Path.dirname` [here](https://github.com/AtomLinter/linter-codeclimate/blob/cc88e576bcf4809cc2fd96e8d7cb7c87a73ad9c8/lib/linter-codeclimate.coffee#L116). This should probably use the directory of the current file if a configuration path isn't able to be found.

**Comment by blimmer** [comment]
@Arcanemagus that's exactly right. I see the deprecation when I'm working in repositories without a `.codeclimate.yml` file and don't when the file is present. thanks for the pointer - it was weird we couldn't repro it in the same env.

**Comment by Arcanemagus** [comment]
Just looked over this again, I'm pretty sure I unintentionally fixed this in #48. 😛

Once that gets released here in a few minutes please double check to make sure it works properly.
